feat: add sound effect for apple consumption (Closes #4)

JuanHPassos opened the pull request:
Adds a sound effect when the snake eats an apple to provide audio feedback and make the game more satisfying.

- Adds the `eat.wav` sound asset.
- Plays the sound on snake/apple collision.

Closes #4

JuanHPassos committed:
code: add sound effect for apple consumption

clr-cera commented on kobra.py:
```diff
@@ -284,9 +287,9 @@ def draw_grid():
     if snake.head.x == apple.x and snake.head.y == apple.y:
-        #snake.tail.append(pygame.Rect(snake.head.x, snake.head.y, GRID_SIZE, GRID_SIZE))
-        snake.got_apple = True;
-        apple = Apple()
```
This line being remove completely breaks the game, the apple is not recreated so its location is not changed, please restore it.

Beside this, I loved the sound!!!!

clr-cera reviewed (changes requested):
Nice PR, just need to fix apple recreation!

JuanHPassos committed:
code: restore apple recreated

clr-cera commented:
Nice! I believe, after conflicts are resolved, it is good to go (and keep it formatted pleaaase, the ci pipeline is not running as this pr was sent before the ci pr was approved)

JuanHPassos committed:
Merge branch 'main' into feat/4/add-eat-sound

clr-cera reviewed (approved):
LGTM!

otaviobia commented:
Looking (sounding?) good!

Before approving, can you confirm we have the rights to use this sound? Was it generated by you (using something like [jsfxr](https://github.com/chr15m/jsfxr)) or taken from a sound library?

(Also, could you change the pr base branch to dev?)

JuanHPassos committed:
Merge branch 'main' into feat/4/add-eat-sound

JuanHPassos committed:
code: switch to public domain audio

I used the first audio from a website that did not have a license allowing its use in this work. The current one is from the site opengameart.org, which has a public domain license.

JuanHPassos committed:
Merge branch 'dev' into feat/4/add-eat-sound

otaviobia reviewed (approved):
LGTM!

LiviaLelis reviewed (approved):
A small nit pick: could you add a readme or licenses file to the assets directory where we can reference where which file has been taken from and their license so we don't loose track of it? 

I'm gonna leave this approved as it looks good already, but I feel like this is an important addition to keep track of the assets we use.

trqt reviewed (changes requested):
This should be moved under a setting, per #68.

JuanHPassos committed:
Merge branch 'dev' into feat/4/add-eat-sound

trqt committed:
Make the gameplay buttery smooth (#58)

* less input lag

* make the snake's movement smooth

JuanHPassos committed:
code: add eat_sound to settings

JuanHPassos committed:
code: black reformatted

JuanHPassos commented:
> This should be moved under a setting, per #68.

Done!

bruno-lim4 committed:
Feat/99/add obstacles (#103)

Add a configurable obstacle system to the naja game that places random static obstacles on the grid, with intuitive difficulty levels that scale based on grid size.

caue-paiva committed:
Feat: Extract entities (Apple,Snake and Obstacles) into a separate file (#107)

caue-paiva committed:
Feat: Create GameState class to encapsulate dynamic game state like the snake,apple, obstacles and arena (#114)

* feat(code-quality): Created entities and constants file to separate the Snake and Apple classes from the main kobra.py game

* create game state class

create game state class

first iteration of state is ready

Game state is working

trying to put this togethe with the other changes to the entities file

* fixing bugs

* fixing merge issues

* Fixed bugs with the obstacles

* linter fixes

* fix comments

LeticiaBN committed:
bug: add 'Reset Game on Apply' toggle to preserve progress (#110)

* fix: add 'Reset Game on Apply' toggle to preserve progress (closes #106)

Added a new setting 'Reset Game on Apply' that allows players to choose
whether to reset the game when applying settings changes during gameplay.

- Default is Off: preserves snake position, progress, and score
- Can be toggled On: resets game when needed (e.g., after grid size changes)
- Players now have control over their gameplay experience
- Fixes issue where all progress was lost when changing settings mid-game

* improve: add smart reset detection for critical settings

- Added '(needs reset)' labels to settings that require game reset
- Implemented automatic reset when critical settings change:
  - Cells per side (grid size)
  - Obstacle difficulty
  - Initial speed
- Game now automatically resets when these settings change, regardless
  of 'Reset Game on Apply' toggle, preventing crashes and inconsistencies
- Safe settings (max speed, sounds, music) can still be changed mid-game
- Better user experience with clear indication of which settings need reset

* style: fix black formatting

* Fix Black formatting for apply_settings call

lfelipediniz committed:
[FIX] Fix death sound timing and add immersive death song (closes #113) (#116)

* doc: add AI assistant documentation for Claude and Cursor

Added comprehensive documentation to help AI coding assistants understand the Naja project structure, conventions, and mechanics.

Files added:
- .claude/CLAUDE.md: Complete project guide for Claude
- .claude/settings.local.json: Claude permission settings
- .cursor/rules/*.mdc: Cursor rule files for code style, workflow, mechanics, and project info

* code: fix death sound playing when game is muted

The gameover sound was being played even when the background music
was muted (MUSIC_ON = False). This happened because the sound playback
only checked DEATH_SOUND_ON but not MUSIC_ON setting.

Now the death sound respects both settings and will not play when
the player has muted the game using the N key.

* code: fix death sound playing after restart instead of on death

The gameover sound was being played after the player pressed restart,
not immediately when the snake died. This happened because the sound
was triggered in the main loop after game_over_handler() returned,
which only happens after the player presses a key to restart.

Moved the sound playback into game_over_handler() so it plays
immediately when the snake dies, before showing the Game Over prompt.
Also removed the unused 'died' variable from the main loop.

* code: add death song that plays during game over screen

Implemented an immersive game over experience with audio transitions:
- When snake dies, play gameover.wav sound effect once
- Immediately stop background music and switch to death_song.mp3
- Death song loops continuously during the Game Over prompt
- When player restarts, death song stops and background music resumes
- When player quits, game exits with death song playing

Added death_song.mp3 (1.7MB) to assets/sound/ directory.
The death song only plays if both DEATH_SOUND_ON and MUSIC_ON are true.

* fix: formated file

Allanvtr committed:
Add opacity to score display (#87)

* code : add opacity to score display

* minor: format code with black

lfelipediniz committed:
Game sounds are now nice and normalized (#119)

* doc: add AI assistant documentation for Claude and Cursor

Added comprehensive documentation to help AI coding assistants understand the Naja project structure, conventions, and mechanics.

Files added:
- .claude/CLAUDE.md: Complete project guide for Claude
- .claude/settings.local.json: Claude permission settings
- .cursor/rules/*.mdc: Cursor rule files for code style, workflow, mechanics, and project info

* code: fix death sound playing when game is muted

The gameover sound was being played even when the background music
was muted (MUSIC_ON = False). This happened because the sound playback
only checked DEATH_SOUND_ON but not MUSIC_ON setting.

Now the death sound respects both settings and will not play when
the player has muted the game using the N key.

* code: fix death sound playing after restart instead of on death

The gameover sound was being played after the player pressed restart,
not immediately when the snake died. This happened because the sound
was triggered in the main loop after game_over_handler() returned,
which only happens after the player presses a key to restart.

Moved the sound playback into game_over_handler() so it plays
immediately when the snake dies, before showing the Game Over prompt.
Also removed the unused 'died' variable from the main loop.

* code: add death song that plays during game over screen

Implemented an immersive game over experience with audio transitions:
- When snake dies, play gameover.wav sound effect once
- Immediately stop background music and switch to death_song.mp3
- Death song loops continuously during the Game Over prompt
- When player restarts, death song stops and background music resumes
- When player quits, game exits with death song playing

Added death_song.mp3 (1.7MB) to assets/sound/ directory.
The death song only plays if both DEATH_SOUND_ON and MUSIC_ON are true.

* fix: formated file

* code: normalize audio volumes and convert to MP3 format

Normalized all audio volumes to 20% for consistent sound levels:
- Background music: 20% volume
- Death song: 20% volume
- Death sound effect: 20% volume (was too loud before)

Converted all audio files to MP3 format for better compatibility:
- BoxCat_Games_CPU_Talk.ogg → BoxCat_Games_CPU_Talk.mp3
- gameover.wav → gameover.mp3
- Updated death_song.mp3 with normalized volume

This ensures all game sounds play at the same comfortable volume level.

trqt committed:
Revert "Game sounds are now nice and normalized (#119)" (#126)

This reverts commit 28c0d2a22a7920b21da6de12a4c69026cf55c063.

LeticiaBN committed:
feat: Add multiple apples support  (#131)

* feat: Add multiple apples support (closes #88)

- Added 'Number of apples' setting to game settings menu
- Updated GameState to manage list of apples instead of single apple
- Modified Snake.update() to accept list of apples
- Game loop now maintains N apples in the arena at all times
- Apples are automatically respawned when eaten (if free cells available)
- Added validation to prevent apple overlap
- Number of apples limited by available free grid squares
- Added 'number_of_apples' to critical settings requiring reset

Implementation details:
- Default: 1 apple (classic behavior)
- Max: 15% of grid cells or 30 apples (whichever is smaller)
- When snake eats apple, new one spawns immediately
- End-game: fills remaining free squares when N > free cells
- All apples properly repositioned on game reset
- Menu cap set to 30 for gameplay balance

* style: Fix Black formatting issues

- Fix line wrapping in entities.py for apple overlap check
- Remove trailing whitespace in state.py
- Remove trailing whitespace in kobra.py

* style: Fix remaining Black formatting issues in kobra.py

- Remove all trailing whitespace
- Wrap long occupied_cells calculation across multiple lines
- Wrap long while condition for apple overlap check

FerSuaiden committed:
solving partially clipped sound indicator (#133)

lfelipediniz committed:
Feat/100/add assets class (#134)

* first iteration of the GameAssets is ready, but there are still some global vars

* main structure ofg the solution is ready, going to fix bugs

* Fixed bugs, code is now ready

* improved type hints.

Co-authored-by: caue-paiva <cauepaivalira@outlook.com>

* fixed lint issues after merge. Co-authored-by: caue-paiva <cauepaivalira@outlook.com>

---------

Co-authored-by: caue-paiva <cauepaivalira@outlook.com>
Co-authored-by: Caue Paiva <128509727+caue-paiva@users.noreply.github.com>

lfelipediniz committed:
Aux/refactor eliminate globals (#136)

* code: implement complete GameSettings class

Transformed GameSettings from a simple stub into a fully functional
class managing game configuration settings and menu fields.

* code: implement complete GameConfig class

Transformed GameConfig from a simple stub into a fully functional
class managing immutable game configuration (screen dimensions,
grid size calculations, and display information).

This refactoring eliminates global variables for display configuration
and follows better programming practices with proper encapsulation.

* code: enhance GameState class with properties and utility methods

Added comprehensive properties for safe dimension access and utility
methods for better state management. Improvements include:
- Properties: width, height, grid_size, cells_per_side, total_cells
- Game control: pause(), resume(), toggle_pause()
- Entity management: add_apple(), remove_apple(), reset_snake()
- Utility methods: get_free_cells_count(), get_state_summary()

This refactoring improves encapsulation and provides a cleaner API
for game state manipulation.

* code: eliminate all global variables from kobra.py

Major refactoring to remove all global variables and follow best practices:

- Replaced global display/config variables with GameConfig class
- Replaced SETTINGS and MENU_FIELDS with GameSettings class
- Updated all functions to receive config and settings as parameters
- Removed all 'global' declarations from the code
- Updated main() to properly initialize and pass class instances
- Used state properties (width, height, grid_size) instead of globals
- Applied better encapsulation and dependency injection patterns

This refactoring improves:
- Code maintainability and testability
- Eliminates hidden dependencies
- Follows SOLID principles
- Makes code more modular and reusable

Stats: 274 insertions(+), 296 deletions(-)

* doc: update inline documentation to English

Improved code documentation by updating comments and docstrings
to English for better consistency and international collaboration.

* code: apply black formatting to constants.py and state.py

* doc: adjust inline documentation and comments

clr-cera reviewed (approved):
LGTM, nice for this to be on settings!

lfelipediniz committed:
repo: organize Python modules into src directory (#139)

JuanHPassos committed:
code: add eat sound

JuanHPassos committed:
code: fix eat sound.

Was missing eat_sound in default settings, and the correct path to the sound file, and change label of eat_sound to english

JuanHPassos committed:
code: resolve conflict

JuanHPassos committed:
code: black reformatted kobra.py

JuanHPassos committed:
code: black reformatted src/settings.py
